woocommerce/woocommerce-ios
# [Card Present Payments] Scaffolding CardReaderService

Closes #3589 

For reference: p91TBi-4q4-p2

This is a very first iteration of the basic abstractions we are going to use to implement the integration with the Stripe Terminal SDK.

Stripe Terminal Reference: https://stripe.dev/stripe-terminal-ios/docs/index.html

It is important to remember that we do not want to mimic the API of the Stripe Terminal SDK, but to provide an adapter wrapping it, so we want to provide methods in the API to:
* start
* cancel
* create payment intents
* process payment intents
* cancel payment intents
* update the reader (checking for updates and updating the reader is intentionally left out of this PR)

What is a payment item? I'm glad you asked! According to [the documentation](https://stripe.dev/stripe-terminal-ios/docs/Classes/SCPPaymentIntent.html):


> A PaymentIntent tracks the process of collecting a payment from your customer. We recommend that you create exactly one PaymentIntent for each order or customer session in your system. You can reference the PaymentIntent later to see the history of payment attempts for a particular session.

> A PaymentIntent transitions through multiple statuses throughout its lifetime and ultimately creates at most one successful charge.


Things to note:
* These are just empty types for now. There is no actual implementation.
* Most of the actual model objects do not declare any properties now. We will get to that later.

## Changes
* Added a protocol, CardReaderService that would be the public facade of our integration with the Stripe Terminal SDK. This protocol is supposed to provide high-level operations and Publishers that clients can observe to inspect the status of the reader.
* Added an empty implementation of the `ConnectionTokenProvider` protocol.
* Added an implementation of the CardReaderService that will end up being the actual integration with the Stripe Terminal SDK. This contains enough (not really functional or useful in any way) code to pass the build.
* Added enough model objects to pass the build. Some of them are empty, some others are enumerations that might be close to their final state.

## How to test
There really isn't much to test. None of this code is executed, but it is something we want to get in as early as possible to start building our prototype (as mentioned in p91TBi-4k7-p2)

Anyway please:
* 👁️  at the code
* Make sure that CI gives the ✅ 

Update release notes:
